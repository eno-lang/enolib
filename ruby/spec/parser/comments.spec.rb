# frozen_string_literal: true

input = <<~DOC.strip
> note
> more notes
    >    note
    >
DOC

describe Enolib::Parser do
  describe 'Comments' do
    context = Enolib::Context.new(input)

    it 'parses as expected' do
      expect(context.document).to match_snapshot
    end
  end
end
