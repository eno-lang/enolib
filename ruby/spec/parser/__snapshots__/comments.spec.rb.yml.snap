---
"./spec/tokenizer/comments.spec.rb[1:1:1]":
- :index: 0
  :line: 0
  :comment: " note"
  :ranges:
    :comment:
    - 1
    - 6
    :comment_operator:
    - 0
    - 1
  :type: :comment
  :length: 6
- :index: 7
  :line: 1
  :comment: " more notes"
  :ranges:
    :comment:
    - 1
    - 12
    :comment_operator:
    - 0
    - 1
  :type: :comment
  :length: 12
- :index: 20
  :line: 2
  :comment: "    note"
  :ranges:
    :comment:
    - 5
    - 13
    :comment_operator:
    - 4
    - 5
  :type: :comment
  :length: 13
- :index: 34
  :line: 3
  :comment: ''
  :ranges:
    :comment:
    - 5
    - 5
    :comment_operator:
    - 4
    - 5
  :type: :comment
  :length: 5
"./spec/parser/comments.spec.rb[1:1:1]":
  :elements: []
  :type: :document
